### ISSUE TITLE
RTransaction.getLocalCachedMap giving wrong cache if the cache map is empty

### ISSUE THREAD

**Issue by raja2102598** [redisson/redisson#5816]
**Steps to reproduce or test case**
```
		RedissonClient redissonClient = createRedissonClient();
		RTransaction transaction = redissonClient.createTransaction(
				TransactionOptions.defaults().timeout(100, TimeUnit.MINUTES));
		RLocalCachedMap<Object, Object> test = redissonClient.getLocalCachedMap(
				LocalCachedMapOptions.name("test"));
		RLocalCachedMap<Object, Object> nextTest = redissonClient.getLocalCachedMap(
				LocalCachedMapOptions.name("nextTest"));
		RLocalCachedMap<Object, Object> testT = transaction.getLocalCachedMap(
				test);
		System.out.println(testT.getName());
		RLocalCachedMap<Object, Object> nextTestT = transaction.getLocalCachedMap(
				nextTest);
		System.out.println(nextTestT.getName());
		transaction.commit();
```
This is output
```
test
test
```

**Redis version**
7.2.4
**Redisson version**
3.29.0
**Redisson configuration**
```
config.setThreads(Integer.parseInt("100"))
				.setCodec(new JsonJacksonCodec(mapper))
				.setNettyThreads(Integer.parseInt("100"))
				.setUseScriptCache(true).useClusterServers()
				.addNodeAddress(redisClusterUrls.split(","))
				.setReadMode(ReadMode.MASTER_SLAVE)
				.setTimeout(Integer.parseInt("5") * 60 * 1000)
				.setConnectTimeout(Integer.parseInt("5") * 60 * 1000)
				.setRetryInterval(5000).setRetryAttempts(6)
				.setSubscriptionsPerConnection(Integer.parseInt("100"))
				.setSubscriptionConnectionPoolSize(Integer.parseInt("1000"))
				.setMasterConnectionPoolSize(Integer.parseInt("1000"))
				.setSlaveConnectionPoolSize(Integer.parseInt("1000"))
```

**Comment by raja2102598** [comment]
@mrniko can you please check this

**Comment by raja2102598** [comment]
#5564 This PR is created for the same issue.

**Comment by raja2102598** [comment]
Fixed with #5829
